### PR TITLE
fs: Add kconfig option for enabling exFAT support

### PR DESCRIPTION
**Pull request by rick1082**
Add Kconfig options for exposing the support of exFAT

Expose the configuration for enable/disable the exFAT support to Kconfig.

Signed-off-by: Jui-Chou Chung <jui-chou.chung@nordicsemi.no>

### CODE REVIEW

File: include/ffconf.h [hunk 1]
```diff
@@ -227,8 +227,11 @@
 /  Instead of private sector buffer eliminated from the file object, common sector
 /  buffer in the file system object (FATFS) is used for the file data transfer. */
 
-
-#define _FS_EXFAT	0
+#if defined(CONFIG_FS_FATFS_EXFAT)
+#define _FS_EXFAT	CONFIG_FS_FATFS_EXFAT
+#else
+#define _FS_EXFAT   0
+#endif
```
Comment (de-nordic) [inline include/ffconf.h:234]: Wouldn't it be enough to just  `_FS_EXFAT` the value of `CONFIG_FS_FATFS_EXFAT`?

Comment (de-nordic) [reply include/ffconf.h:234]: OK my bad. It is common file.

Comment (rick1082) [reply include/ffconf.h:234]: There could be a chance that user doesn't defined CONFIG_FS_FATFS_EXFAT, so we might need to remain _FS_EXFAT in default setting as 0.

**Comment by carlescufi** [comment]
> I think If we start using exFAT the worth to consider upgrade of this module - see [zephyrproject-rtos/zephyr#13276 (comment)](https://github.com/zephyrproject-rtos/zephyr/issues/13276#issuecomment-465156269)
> 
> upgrades introduces a few exFAT issues.

Agreed, let's take the opportunity to update the module as well

**Pull request merged**